tools/SourceKit/tools/sourcekitd-test/sourcekitd-test.cpp

```diff
@@ -1934,6 +1949,26 @@ static void printExpressionType(sourcekitd_variant_t Info, llvm::raw_ostream &OS
   OS << "</ExpressionTypes>\n";
 }
 
+static void printVariableType(sourcekitd_variant_t Info, llvm::raw_ostream &OS) {
+  auto TypeBuffer = sourcekitd_variant_dictionary_get_value(Info, KeyVariableTypeList);
+  unsigned Count = sourcekitd_variant_array_get_count(TypeBuffer);
+  if (!Count) {
+    OS << "cannot find variable types in the file\n";
+    return;
+  }
+  OS << "<VariableTypes>\n";
+  for (unsigned i = 0; i != Count; ++i) {
+    sourcekitd_variant_t Item = sourcekitd_variant_array_get_value(TypeBuffer, i);
+    unsigned Offset = sourcekitd_variant_dictionary_get_int64(Item, KeyVariableOffset);
+    unsigned Length = sourcekitd_variant_dictionary_get_int64(Item, KeyVariableLength);
+    bool HasExplicitType = sourcekitd_variant_dictionary_get_bool(Item, KeyVariableTypeExplicit);
+    OS << "(" << Offset << ", " << Offset + Length << "): "
```

Comment:
I’m not sure what you think about this, but I think it would be easier to write tests if we were using line-column based positions here instead of offset-based ones. You can convert between the two using `resolveToLineCol`.

Reply:
I feel like offset-based positions are more in line with the request itself (which uses offset + length), slightly more compact and not much harder to test. But I would be fine either way.

Reply:
I just thought that the tests read better, because e.g. [here](https://github.com/apple/swift/pull/37867/files#diff-2a7ab5b5b0f42a4779d38a6699a951ec6013673074023b669f65cac2699dc11dR32) I can’t tell at a glance what offset `230`, whereas `(19:6, 19:11)` makes it clear at first glance why variable we’re referring to.